### PR TITLE
Fixes unsafe reflection warning in dashboards_controller

### PR DESCRIPTION
**Pull request by diogoosorio**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This one is purely to make CodeClimate/brakeman happy as the code that was already there is equivalent (as the parameter being `constantize`'d was validated before the reflection part kicked in).

I've switch things around and do a lookup on the `UserSubscription::ALLOWED_TYPES` for the subscription `source_type`, which makes brakeman understand that type comes from an allowed list of types (instead the user input).

## Related Tickets & Documents

#3739

## QA Instructions, Screenshots, Recordings

Guarantee that no regression was introduced (the code is indeed equivalent):

1. The `requests/dashboard_spec.rb` covers these scenarios

---

1. Visit: http://localhost:3000/dashboard/subscriptions?source_type=Article&source_id=1
1. See that the page loads the subscriptions of the correct article

---

1. Visit: http://localhost:3000/dashboard/subscriptions?source_type=Comment&source_id=1
1. See that that a 404 page is rendered

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by citizen428** [comment]
Thanks for this PR! I'm not a big fan of changing code to please a tool, especially since the `detect` version is just writing out explicitly what `include?` does. But I have no problem merging this if others are ok with it.

File: app/controllers/dashboards_controller.rb [hunk 1]
```diff
@@ -76,8 +76,9 @@ def subscriptions
   private
 
   def set_source
-    source_type = params[:source_type]
-    not_found unless UserSubscription::ALLOWED_TYPES.include? source_type
+    source_type = UserSubscription::ALLOWED_TYPES.detect { |allowed_type| allowed_type == params[:source_type] }
+
+    not_found if source_type.nil?
```
Comment (rhymes) [inline app/controllers/dashboards_controller.rb:81]:
```suggestion
    not_found unless source_type
```

**Review by rhymes — COMMENTED** [review]
I'd change the `if` to an `unless` but other than that I agree that CC is overzealous here but as it's not breaking anything, we can merge it

**Review by rhymes — APPROVED** [review]
Thanks @diogoosorio

**Pull request merged**